src/xrc/src/api/test.rs

```diff
@@ -1662,3 +1662,295 @@ mod timestamp_is_in_future {
         );
     }
 }
+
+mod request_contains_invalid_symbols {
+
+    use super::*;
+
+    /// This function tests that a crypto pair request with an invalid base asset symbol
+    /// is rejected and charged the minimum fee.
+    #[test]
+    fn handle_cryptocurrency_pair_invalid_base_asset_symbol() {
+        let current_timestamp: u64 = 1678752000;
+        let call_exchanges_impl = TestCallExchangesImpl::builder().build();
+        let env = TestEnvironment::builder()
+            .with_time_secs(current_timestamp)
+            .with_cycles_available(XRC_REQUEST_CYCLES_COST)
+            .with_accepted_cycles(XRC_MINIMUM_FEE_COST)
+            .build();
+        let request = GetExchangeRateRequest {
+            base_asset: Asset {
+                symbol: "<>".to_string(),
+                class: AssetClass::Cryptocurrency,
+            },
+            quote_asset: Asset {
+                symbol: "ICP".to_string(),
+                class: AssetClass::Cryptocurrency,
+            },
+            timestamp: Some(current_timestamp),
+        };
+
+        let result = get_exchange_rate_internal(&env, &call_exchanges_impl, &request)
+            .now_or_never()
+            .expect("future should complete");
+        assert!(matches!(
+            result,
+            Err(ExchangeRateError::CryptoBaseAssetNotFound)
+        ));
+    }
+
+    /// This function tests that a crypto pair request with an invalid quote asset symbol
+    /// is rejected and charged the minimum fee.
+    #[test]
+    fn handle_cryptocurrency_pair_invalid_quote_asset_symbol() {
+        let current_timestamp: u64 = 1678752000;
+        let call_exchanges_impl = TestCallExchangesImpl::builder().build();
+        let env = TestEnvironment::builder()
+            .with_time_secs(current_timestamp)
+            .with_cycles_available(XRC_REQUEST_CYCLES_COST)
+            .with_accepted_cycles(XRC_MINIMUM_FEE_COST)
+            .build();
+        let request = GetExchangeRateRequest {
+            base_asset: Asset {
+                symbol: "ICP".to_string(),
+                class: AssetClass::Cryptocurrency,
+            },
+            quote_asset: Asset {
+                symbol: "<>".to_string(),
+                class: AssetClass::Cryptocurrency,
+            },
+            timestamp: Some(current_timestamp),
+        };
+
+        let result = get_exchange_rate_internal(&env, &call_exchanges_impl, &request)
+            .now_or_never()
+            .expect("future should complete");
+        assert!(matches!(
+            result,
+            Err(ExchangeRateError::CryptoQuoteAssetNotFound)
+        ));
+    }
+
+    /// This function tests that a crypto/fiat pair request with an invalid base asset symbol
```

Comment:
The asset class doesn't have any effect here. Is there a benefit in testing all possible scenarios?

Reply:
To ensure total coverage, we have to go through all scenarios. The `validate_request` function is used in each `handle_` function. At some point, I would like to figure out a way to refactor it to be in a singular point, but I would also like to avoid having many places where we could charge for cycles.